apps/server/src/modules/authentication/strategy/x-api-key.strategy.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ConfigService } from '@nestjs/config';
+import Strategy from 'passport-headerapikey';
+import { XApiKeyConfig } from '../config/x-api-key.config';
+
+@Injectable()
+export class XApiKeyStrategy extends PassportStrategy(Strategy, 'api-key') {
+	private readonly allowedApiKeys: string[];
+
+	constructor(private readonly configService: ConfigService<XApiKeyConfig, true>) {
+		super({ header: 'X-API-KEY' }, false);
+		this.allowedApiKeys = this.configService.get<string[]>('ADMIN_API__ALLOWED_API_KEYS');
+	}
+
+	public validate = (apiKey: string, done: (error: Error | null, data: boolean | null) => void) => {
```

Comment:
The x-api-key must match to "real" (system) user that has real permissions, that can be authorised. This userId can be used for the logging. For the authorisation a rule must be exists. A strategy like you has the key and a unkown can do all, is not what we want.

Reply:
This is correct, but in this case (at this moment) it's only used in the "system2system" communication, there is no real user in a sense there is no one that could have this or that permission to do that, this API will be called by an automatic CronJob for now. And as there is no other solution present in the SC to do this kind of an authentication we've added the X-API-Key to have **any** authentication layer in this communication.

Reply:
@CeEv please let us know if this argument is enough for now, if not I guess we'll have to meet and discuss it.